pkg/es/config/config.go

```diff
@@ -238,20 +248,20 @@ func (c *Configuration) ApplyDefaults(source *Configuration) {
 	if c.AdaptiveSamplingLookback == 0 {
 		c.AdaptiveSamplingLookback = source.AdaptiveSamplingLookback
 	}
-	if c.NumShards == 0 {
-		c.NumShards = source.NumShards
+	if c.Indices.Dependencies.TemplateOptions.NumShards == 0 {
```

Comment:
previously shards/replicas settings were defined once for all indices. You now partitioned them by index type - a good feature, but now you need to handle them all separately.

Reply:
So your suggestion is to keep shards/replicas as the global index's shards/replicas settings?

Reply:
No, I like the new change as it makes things more flexible and not too cumbersome to users if we have sensible defaults. My point was that the code needs to be changed accordingly to account for splitting previously 2 global values into 4 independent groups of values.